src/spetlr/cache/CachedLoader.py

```diff
@@ -101,7 +101,16 @@ def save(self, df: DataFrame) -> None:
                 return
 
         # write branch
-        df_written = self.write_operation(result.to_be_written)
+        if self.params.provisional_markup_step:
+            pre_write_version = self._perform_provisional_markup(result.to_be_written)
+            df_written = self.write_operation(result.to_be_written)
+            # no context manager or try-catch loop is used here. If the operation
+            # fails, we want the provisional markup to remain, ensuring correct data
+            # updates on next run.
+            self._rollback_provisional_markup(pre_write_version)
```

Comment:
We may have a chance for substantial optimization if we do not allow the child class to filter the result. Then we do not need to roll back the provisional makup

Reply:
On further investigation, I do not see any substantial optimization this way. We may however optimize substantially if we partition the cache, e.g. by date or week, and then pass the filter to only export, say, the last 10 weeks' worth of data. That way we can prevent the cache from growing as time goes by. I am now tracking this in #207